terraform/provider/resource_teleport_bot.go

```diff
@@ -0,0 +1,241 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provider
+
+import (
+	"context"
+	"time"
+
+	"github.com/gravitational/teleport-plugins/terraform/tfschema"
+	"github.com/gravitational/teleport/api/client/proto"
+	"github.com/gravitational/teleport/api/types/wrappers"
+	"github.com/gravitational/trace"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+func GenSchemaBot(ctx context.Context) (tfsdk.Schema, diag.Diagnostics) {
+	return tfsdk.Schema{
+		Attributes: map[string]tfsdk.Attribute{
+			"id": {
+				Type:     types.StringType,
+				Computed: true,
+			},
+			"name": {
+				Type:        types.StringType,
+				Required:    true,
+				Description: "The name of the bot, i.e. the unprefixed User name",
+				PlanModifiers: []tfsdk.AttributePlanModifier{
+					tfsdk.RequiresReplace(),
+				},
+			},
+			"user_name": {
+				Type:        types.StringType,
+				Computed:    true,
+				Description: "The name of the generated bot user",
+			},
+			"role_name": {
+				Type:        types.StringType,
+				Computed:    true,
+				Description: "The name of the generated bot role",
+			},
+			"token_ttl": {
+				Type:        types.StringType,
+				Optional:    true,
+				Computed:    true,
+				Description: "The desired TTL for the token if one is created. If unset, a server default is used",
+			},
+			"token_id": {
+				Type:        types.StringType,
+				Optional:    true,
+				Computed:    true,
+				Description: "The bot joining token. If unset, a new random token is created and its name returned, otherwise a preexisting Bot token may be provided for IAM/OIDC joining.",
+
+				// TODO: Consider dropping RequiresReplace() in the future if a
+				// ResetBot() API becomes available.
```

Comment:
Can we create an issue outlining what `ResetBot()` would do so we just keep track of this as a piece of work we'd want to engage with in future ?

Reply:
This is covered in https://github.com/gravitational/teleport/issues/13091, I'll link it in a comment here